src/index.ts

```diff
@@ -195,11 +197,9 @@ export async function createWithExpiration (peerId: PeerId, value: CID | PeerId
 export async function createWithExpiration (peerId: PeerId, value: CID | PeerId | string, seq: number | bigint, expiration: string, options: CreateOptions = defaultCreateOptions): Promise<IPNSRecord> {
   const expirationDate = NanoDate.fromString(expiration)
   const validityType = IpnsEntry.ValidityType.EOL
+  const lifetimeNs = typeof options.lifetimeNs === "bigint" ? options.lifetimeNs : DEFAULT_TTL
```

Comment:
If set and not a bigint, should we error or convert instead of silently overriding?

Reply:
Would throwing an error by attempting to cast the value to BigInt while checking for a typeof undefined work?
`const lifetimeNs = typeof options.lifetimeNs !== "undefined" ? BigInt(options.lifetimeNs) : DEFAULT_TTL`